flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/strategy/InputDependencyConstraintCheckerTest.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.strategy;
+
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.apache.flink.api.common.InputDependencyConstraint.ALL;
+import static org.apache.flink.runtime.io.network.partition.ResultPartitionType.PIPELINED;
+import static org.apache.flink.runtime.scheduler.strategy.SchedulingResultPartition.ResultPartitionState.EMPTY;
+import static org.apache.flink.runtime.scheduler.strategy.SchedulingResultPartition.ResultPartitionState.PRODUCING;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Unit tests for {@link InputDependencyConstraintChecker}.
+ */
+public class InputDependencyConstraintCheckerTest extends TestLogger {
+
+	@Test
+	public void testCheckInputVertex() {
+		final TestingSchedulingTopology testingSchedulingTopology = new TestingSchedulingTopology();
+
+		final List<TestingSchedulingExecutionVertex> producers = testingSchedulingTopology.addExecutionVertices().finish();
+
+		final InputDependencyConstraintChecker inputChecker = testingSchedulingTopology
```

Comment:
I think it's better to create the `InputDependencyConstraintChecker` instance without `TestingSchedulingTopology` because:
- there are no complex relationships with other objects, such as `SchedulingExecutionVertex`
- InputDependencyConstraintChecker is only used inside this test but not in other tests

A static factory method `createInputDependencyConstraintChecker(List<? extends SchedulingResultPartition> resultPartitions)` inside the unit test should be enough.

Reply:
OK, involving `TestingSchedulingTopology` into `InputDependencyConstraintChecker` is not a good choice.

As I would like to test various combinations of `ResultPartitionType`, `ResultPartitionState` and `InputDependencyConstraint`, I add simple `TestingSchedulingExecutionVertexBuilder` and `TestingSchedulingResultPartitionBuilder` in `InputDependencyConstraintCheckerTest`.